pkg/cmd/roachtest/tests/rust_helpers.go

```diff
@@ -0,0 +1,67 @@
+package tests
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"regexp"
+)
+
+var rustUnitTestOutputRegex = regexp.MustCompile(`(?P<type>.*) (?P<class>.*)::(?P<name>.*) .\.\.\ (?P<result>.*)`)
+
+func (r *ormTestsResults) parseRustUnitTestOutput(
```

Comment:
this looks very similar to `parsePythonUnitTestOutput` barring a few words. is it possible to add parameters and make them both reference the same impl,

e.g.

```

func (r *ormTestsResults) parseUnitTestOutput(input []byte, expectedFailures blocklist, ignoreList blocklist, outputRe *regexp.Regexp, resultName string) {
   // replace ["reason"] and rustUnitTestOutputRegex here
}

func (r *ormTestsResults) parseRustUnitTestOutput(input []byte, expectedFailures blocklist, ignoreList blocklist) {
   return r.parseUnitTestOutput(input, expectedFailure, ignoreList, re, resultName)
}

func (r *ormTestsResults) parsePythonUnitTestOutput(input []byte, expectedFailures blocklist, ignoreList blocklist) {
   return r.parseUnitTestOutput(input, expectedFailure, ignoreList, re, resultName)
}
```

if you feel there's more differences than that, please feel free to skip this suggestion.